pkg/patterns/declarative/reconciler.go

```diff
@@ -198,42 +236,41 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 	objects, err := r.BuildDeploymentObjectsWithFs(ctx, name, instance, fs)
 	if err != nil {
 		log.Error(err, "building deployment objects")
-		return nil, fmt.Errorf("error building deployment objects: %v", err)
+		return statusInfo, fmt.Errorf("error building deployment objects: %v", err)
 	}
+	statusInfo.Manifest = objects
```

Comment:
Super nit:
The logic is a little bit confusing considerin that [ln222/259](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/272/files#diff-91d9000f2817ed2698caf7b294e34ed1bc739bf60b7dc1af3136055b18741312R259) `objects, err = parseListKind(objects)` and then 
[ ln207/264](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/272/files#diff-91d9000f2817ed2698caf7b294e34ed1bc739bf60b7dc1af3136055b18741312R264) has duplicate `statusInfo.Manifest = objects`

Maybe moving the logic[ ln259-264](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/272/files#diff-91d9000f2817ed2698caf7b294e34ed1bc739bf60b7dc1af3136055b18741312R259-R264) here. Otherwise it looks like a duplication of  ln 264 and not obvious enough why the code appends `Objects` to Manfiests multiple times.

Reply:
You're right - this logic is super confusing.  I think it has "evolved" over time and some entropy has snuck in, and we should clean it up.

I will send a separate PR if that's OK - your point is very valid but I think it's a separate concern to the logic in this PR (i.e. this problem already existed)